File: ProblemSet/fhoj/1106/1106.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int main (int argc, char const* argv[])
{
    int n, flg, i;

    scanf("%d", &n);

    flg = 1;
    for (i = 2; i <= (int)(sqrt(n)); i++) {
        if (n % i == 0) {
            flg = 0; break;
        }
    }

    if (flg) {
        printf("Yes\n");
    } else {
        printf("No\n");
    }

    return 0;
}

File: ProblemSet/fhoj/1106/1106.pas
program q1106;
var
    n, flg, i: longint;
begin
    assign(input, '');
    assign(output, '');
    reset(input);
    rewrite(output);

    readln(n);

    flg := 1;
    for i:=2 to trunc(sqrt(n)) do
        if (n mod i = 0) then
            flg := 0;

    if (flg = 1) then
        writeln('Yes')
    else
        writeln('No');

    close(input);
    close(output);
end.
